databricks_cli/dbfs/api.py

```diff
@@ -83,17 +91,30 @@ class DbfsErrorCodes(object):
     TOO_MANY_REQUESTS = 'TOO_MANY_REQUESTS'
 
 
+def before_sleep_on_429(retry_state):
+    global time_for_last_retry
+    if retry_state.attempt_number < 1:
+        loglevel = logging.INFO
+    else:
+        loglevel = logging.WARNING   
+    logger.log(
+        loglevel, ' Received 429 REQUEST_LIMIT_EXCEEDED for attempt %s. Retrying in %s seconds.',
+        retry_state.attempt_number, retry_state.idle_for - time_for_last_retry)
+    time_for_last_retry = retry_state.idle_for
```

Comment:
This is not going to work as expected in the case where more than 1 call to functions with retry_429 happen (e.g. in the case of `put_file`). The problem is that `time_for_last_retry` is only initialised once per databricks-cli command, while it should be initialised every time a function with retry_429 is called, in the first attempt.

A quick fix that will work (only if code is single-threaded! which is the case here) is initialising `time_for_last_retry` if `retry_state.attempt_number == 1` (I don't think `attempt_number` can be lower than 1, or at least that's what I noticed from testing).

Anyway, I will merge this PR as it is and address this comment myself in a separate PR.